### ISSUE TITLE
RFE: allow specifier expansion for .include directive in unit files

### ISSUE THREAD

**Issue by AnoopAlias** [systemd/systemd#5881]
- [* ] Request for enhancement (RFE)
systemd 219  (CentOS7)

Instantiated unit 

```
[Unit]
DefaultDependencies=no

[Service]
User=%i
Group=nobody
Environment="FPM_SOCKETS=/opt/php70/root/var/run/%i.sock=3"
ExecStart=/opt/php70/root/usr/sbin/php-fpm --prefix=/opt/php70/root --fpm-config=/opt/secure-php-fpm.d/%i.conf
KillMode=process
.include /opt/limits.d/%i.resource

```

The include file has

```
[Service]
Slice=limits.slice

```

Everything works if the .include is an obsolute path 

eg: /opt/limits.d/user.resource

But not when I use %i  in the filename .

It would be great to have ability include different files based on different value of the %i

**Comment by poettering** [comment]
".include" has been depreacted since a long time and removed from the documentation. At this point we only support it for compatibility, and will not add any new features to it.

Sorry.

**Comment by AnoopAlias** [comment]
Ok what about .d includes

I tried creating

/etc/systemd/system/php70@.service.d/user.d/resource.conf

and 

/etc/systemd/system/php70.service.d/user.d/resource.conf

But looks like
```
[Service]
Slice=limits.slice

```
 in the resource.conf is not included 

Basically, I want to start different instantiated units from a different slice for resource control.
As mentioned earlier if the file in .include is an absolute one this works and the service is under the limits.slice .

**Comment by poettering** [comment]
If you want too add a drop-in for unit file "foo.service", then place it in /etc/systemd/system/foo.service.d/50-waldo.conf", or suchlike.

Or in other words: there's not supposed to be a "user.d" in your path, where did you get that from?

But please note, this is a bug tracker, not a support forum. Please contact the systemd mailing list if you just have questions how to use things. Thank you for your understanding.

**Comment by AnoopAlias** [comment]
I got it from:
https://www.freedesktop.org/software/systemd/man/systemd.unit.html

> Along with a unit file foo.service, a "drop-in" directory foo.service.d/ may exist. All files with the suffix ".conf" from this directory will be parsed after the file itself is parsed. This is useful to alter or add configuration settings for a unit, without having to modify unit files. Each drop-in file must have appropriate section headers. Note that for instantiated units, this logic will first look for the instance ".d/" subdirectory and read its ".conf" files, followed by the template ".d/" subdirectory and the ".conf" files there. 


I am trying to do a drop in file for instantiated unit

**Comment by poettering** [comment]
that man page doesn't mention "user.d" anywhere? somehow you added one component too much to your path

**Comment by bmeh** [comment]
I know this is old, and sorry for bumping it but I've just noticed this warning today in dmesg:

```
systemd[1]: /etc/systemd/system/netctl@ethernet\x2dstatic.service:1: .include directives are deprecated, and support for them will be removed in a future version of systemd. Please use drop-in files instead.
```

What am I exactly supposed to do? Move `/usr/lib/systemd/system/netctl@.service` to `/etc/systemd/system/netctl\@ethernet\\x2dstatic.d/00-netctl.conf` after creating that directory? What if other, non-ethernet profiles want to use it as well? Am I supposed to create a `*.d` directory and create duplicates?

I found this: `/usr/bin/netctl:    echo ".include /usr/lib/systemd/system/netctl@.service" > "$unit"` so I guess they'll do it automatically when `.include` will eventually won't even be supported (?) but is the solution really duplication? What's the correct solution in such a scenario? Maybe I'll forward it to netctl, but I'm curious as to what @poettering would do.

Thanks for your response in advance!
